pywinauto/controls/uiawrapper.py

```diff
@@ -420,6 +420,11 @@ def invoke(self):
         """An interface to the Invoke method of the Invoke control pattern"""
         self.iface_invoke.Invoke()
 
+        name = self.element_info.name
+        control_type = self.element_info.control_type
```

Comment:
Invoking button could close the dialog (and crash this code) so it's better to get name and control_type before calling action.